libfreerdp/common/settings.c

```diff
@@ -1570,7 +1570,7 @@ void* freerdp_settings_get_pointer_array_writable(const rdpSettings* settings, s
 			max = freerdp_settings_get_uint32(settings, FreeRDP_TargetNetAddressCount);
 			if (offset >= max)
 				goto fail;
-			return settings->TargetNetPorts[offset];
+			return (void*)(uintptr_t)settings->TargetNetPorts[offset];
```

Comment:
Is this really what we want?
`TargetNetPorts` is a `UINT32*` so we would return the Port value as an address the way the code is currently

Reply:
@mfleisz no, `TargetNetPorts[x]` is a `UINT32`. we return the value in a `void*` (see function return). To avoid any issues with integer/pointer sizes, it is first cast to `uintptr_t` so that no invalid data is there if e.g. `uintptr_t` or `size_t ` integer variables are used on the other end.

Reply:
@akallabeth from `include/freerdp/settings.h`:

` ALIGN64 UINT32* TargetNetPorts;`

Reply:
@akallabeth right - but why do we return the port value in a `void*`? The function `freerdp_settings_get_pointer_array_writable` suggests that I get a pointer that I am able to write to. Seems dangerous if i.e. we return `(void*)3389` from that function ...

Reply:
@mfleisz but good point, this function is not used with `FreeRDP_TargetNetPorts` currently, so change the return to a `UINT32*` value